File: core/src/main/java/com/ustadmobile/core/view/XapiPackageContentView.java
package com.ustadmobile.core.view;

import com.ustadmobile.core.impl.UmCallback;

/**
 * Created by mike on 9/13/17.
 */

public interface XapiPackageContentView extends UstadView, ViewWithErrorNotifier{

    String VIEW_NAME = "XapiPackage";

    String ARG_CONTAINER_UID = "containerUid";

    void setTitle(String title);

    void loadUrl(String url);

    void mountContainer(long containerUid, UmCallback<String> callback);

}

File: core/src/main/java/com/ustadmobile/core/view/XapiPackageContentView.kt
package com.ustadmobile.core.view

import com.ustadmobile.core.impl.UmCallback

/**
 * Created by mike on 9/13/17.
 */

interface XapiPackageContentView : UstadView, ViewWithErrorNotifier {

    fun setTitle(title: String)

    fun loadUrl(url: String)

    fun mountContainer(containerUid: Long, callback: UmCallback<String>?)

    companion object {

        val VIEW_NAME = "XapiPackage"

        val ARG_CONTAINER_UID = "containerUid"
    }

}
